### PR TITLE
Add support for markdown file input and update documentation

### PR DESCRIPTION
**Pull request by tkhongsap**
This Pull Request introduces the following changes:

1. **Markdown File Support**:
   - Added functionality to handle markdown file input for the content extraction process.
   - Updated `content_extractor.py` and `markdown_extractor.py` to support parsing of markdown files.
   
2. **Tests**:
   - Added unit tests in `tests/test_markdown_extractor.py` to ensure that markdown extraction works correctly.

3. **Documentation Updates**:
   - Updated the usage documentation to include information about markdown file input.
   - Updated `README.md` and usage guides to reflect the new functionality.

4. **Miscellaneous**:
   - Updated `pyproject.toml` and `requirements.txt` to include any new dependencies related to markdown support.

Please review the changes, and let me know if any updates or adjustments are required.

### CODE REVIEW

**Comment by souzatharsis** [comment]
This looks great and nicely organized and documented.

However, automated tests did not pass. Could you please review?

**Comment by tkhongsap** [comment]
Okay, will check again. I ran test on the markdown extractor, it passed.

tests\test_markdown_extractor.py ...                                                                                                    [100%]
============================================================= 3 passed in 0.24s =============================================================

**Comment by souzatharsis** [comment]
We need to make sure all tests in test/*.py pass to avoid regressions.

**Comment by souzatharsis** [comment]
Please see pytest logs at https://github.com/souzatharsis/podcastfy/actions/runs/11384115244/job/31671236157?pr=79

**Comment by tkhongsap** [comment]
Okay, thank you for pointing that out. Will take a look.

File: podcastfy/client.py [hunk 1]
```diff
@@ -71,14 +71,20 @@ def process_content(
...
-                api_key=config.GEMINI_API_KEY, conversation_config=conv_config.to_dict()
+                api_key=config.OPENAI_API_KEY, conversation_config=conv_config.to_dict()
             )
 
             if urls:
                 logger.info(f"Processing {len(urls)} links")
                 content_extractor = ContentExtractor()
-                # Extract content from links
-                contents = [content_extractor.extract_content(link) for link in urls]
+                # Extract content from links or file paths
```
Comment (souzatharsis) [inline podcastfy/client.py:80]: it's better to add the below logic (test whether it's a link or file path) to inside extract_content and keep this as list comprehension

File: podcastfy/config.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ output_directories:
   audio: "./data/audio"
 
 content_generator:
-  gemini_model: "gemini-1.5-pro-latest"
+  openai_model: "gpt-4o-mini"
```
Comment (souzatharsis) [inline podcastfy/config.yaml:6]: let's keep gemini as the default.

gpt-4o-mini has not been tested extensively
we use gemini 1.5 pro due to its massive context window of 2M token which allow for processing a list of input sources without trouble

we should consider generalizing to other llm backends here in a separate PR, we should use langchain for that purpose inside of implementing one backend at a time

File: podcastfy/content_generator.py [hunk 3]
```diff
@@ -18,6 +18,7 @@
 from podcastfy.utils.config import load_config
 import logging
 from langchain.prompts import HumanMessagePromptTemplate
+from langchain_openai import ChatOpenAI
```
Comment (souzatharsis) [inline podcastfy/content_generator.py:21]: see above comment

File: podcastfy/content_generator.py [hunk 4]
```diff
@@ -48,10 +49,10 @@ def __init__(
         if is_local:
             self.llm = Llamafile()
         else:
-            self.llm = ChatGoogleGenerativeAI(
-                model=model_name,
+            self.llm = ChatOpenAI(
```
Comment (souzatharsis) [inline podcastfy/content_generator.py:52]: see above comment

File: podcastfy/content_generator.py [hunk 5]
```diff
@@ -63,10 +64,10 @@ def __init__(
         Initialize the ContentGenerator.
 
         Args:
-                api_key (str): API key for Google's Generative AI.
+                api_key (str): API key for OpenAI.
```
Comment (souzatharsis) [inline podcastfy/content_generator.py:67]: see above comment

File: podcastfy/content_generator.py [hunk 6]
```diff
@@ -63,10 +64,10 @@ def __init__(
...
                 conversation_config (Optional[Dict[str, Any]]): Custom conversation configuration.
         """
-        os.environ["GOOGLE_API_KEY"] = api_key
+        os.environ["OPENAI_API_KEY"] = api_key
```
Comment (souzatharsis) [inline podcastfy/content_generator.py:70]: same here

File: podcastfy/content_generator.py [hunk 7]
```diff
@@ -210,9 +211,9 @@ def main(seed: int = 42, is_local: bool = False) -> None:
     """
     try:
         config = load_config()
-        api_key = config.GEMINI_API_KEY if not is_local else ""
+        api_key = config.OPENAI_API_KEY if not is_local else ""
```
Comment (souzatharsis) [inline podcastfy/content_generator.py:214]: and here

File: podcastfy/content_generator.py [hunk 8]
```diff
@@ -210,9 +211,9 @@ def main(seed: int = 42, is_local: bool = False) -> None:
...
         if not is_local and not api_key:
-            raise ValueError("GEMINI_API_KEY not found in configuration")
+            raise ValueError("OPENAI_API_KEY not found in configuration")
```
Comment (souzatharsis) [inline podcastfy/content_generator.py:216]: and here

File: podcastfy/content_parser/content_extractor.py [hunk 9]
```diff
@@ -62,15 +66,17 @@ def extract_content(self, source: str) -> str:
 			ValueError: If the source type is unsupported.
 		"""
 		try:
-			if self.is_url(source):
```
Comment (souzatharsis) [inline podcastfy/content_parser/content_extractor.py:65]: it's better to have a dedicated method for that as self.is_url, curious the motivation to remove

File: podcastfy/conversation_config.yaml [hunk 10]
```diff
@@ -11,7 +11,7 @@ dialogue_structure:
   - "Conclusion"
 podcast_name: "PODCASTFY"
 podcast_tagline: "Your Personal Generative AI Podcast"
-output_language: "English"
+output_language: "Thai"
```
Comment (souzatharsis) [inline podcastfy/conversation_config.yaml:14]: please keep English as default

File: pyproject.toml [hunk 11]
```diff
@@ -44,23 +44,5 @@ types-pyyaml = "^6.0.12.20240917"
...
-[build-system]
-requires = ["poetry-core"]
-build-backend = "poetry.core.masonry.api"
-
-[tool.poetry.scripts]
-build_docs = "build_docs:main"
-
-
-
+langchain-openai = "^0.2.2"
```
Comment (souzatharsis) [inline pyproject.toml:47]: please drop openai for now

File: requirements.txt [hunk 12]
```diff
@@ -67,11 +67,13 @@ langchain-community==0.3.2 ; python_version >= "3.11" and python_version < "4.0"
 langchain-core==0.3.11 ; python_version >= "3.11" and python_version < "4.0"
 langchain-google-genai==2.0.1 ; python_version >= "3.11" and python_version < "4.0"
 langchain-google-vertexai==2.0.5 ; python_version >= "3.11" and python_version < "4.0"
+langchain-openai==0.2.2 ; python_version >= "3.11" and python_version < "4.0"
```
Comment (souzatharsis) [inline requirements.txt:70]: please drop openai for now

File: template_output.md
Comment (souzatharsis) [inline template_output.md:1]: is this file required?

File: test-get-template.py
Comment (souzatharsis) [inline test-get-template.py:1]: is this required?

File: test-podcastfy.py [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+# # Print formatted output with bold blue text
+# print(f"{Fore.BLUE}{Style.BRIGHT}Podcast generated successfully: {audio_file}")
+
+import os
+import subprocess
+
+def run_command(command):
+    subprocess.run(command, shell=True, check=True)
+
+def main():
```
Comment (souzatharsis) [inline test-podcastfy.py:30]: you can simply run make gen-doc in bash instead. there is a Makefile in root that does the below for you

File: usage/config.md [hunk 14]
```diff
@@ -9,12 +9,12 @@ The project uses a combination of a `.env` file for managing API keys and sensit
...
    ```
    GEMINI_API_KEY=your_gemini_api_key_here
-   ELEVENLABS_API_KEY=your_elevenlabs_api_key_here
    OPENAI_API_KEY=your_openai_api_key_here
+   ELEVENLABS_API_KEY=your_elevenlabs_api_key_here
    ```
 API Key Requirements:
-- GEMINI_API_KEY: Required for transcript generation if not using a [local llm](local_llm.md). (get your [free API key](aistudio.google.com/app/apikey))
-- OPENAI_API_KEY or ELEVENLABS_API_KEY: Required for audio generation if not using Microsoft Edge TTS `tts_model=edge`.
+- OPENAI_API_KEY: Mandatory for all operations. (get your API key from [OpenAI](https://platform.openai.com/account/api-keys))
```
Comment (souzatharsis) [inline usage/config.md:16]: please drop openai for now

**Review by souzatharsis — CHANGES_REQUESTED** [review]
Thanks for your PR. I've added comments across several files.

The core observation is that you are replacing ContentGeneration llm backend from Gemini with OpenAI. We should deal with generalizing backend as a separate PR. This one should be focused on markdown support.

Thanks!

File: podcastfy/config.yaml (hunk 2 above)
Comment (tkhongsap) [reply podcastfy/config.yaml:6]: Thank you for pointing that out. I was testing on my local machine and didn’t switch it back. The reason I changed from Gemini to OpenAI was for extracting the "Thai" language, as OpenAI seemed to provide better translations in terms of context. However, it's not an issue—I’ll switch it back.

**Comment by tkhongsap** [comment]
Understood and will do, :-)

**Comment by tkhongsap** [comment]
Understood and will do.
